CharlieBackend.Business/Services/AttachmentService.cs

```diff
@@ -89,20 +90,25 @@ public async Task<Result<IList<AttachmentDto>>> GetAttachmentsListAsync(Attachme
             long accountId = _currentUserService.AccountId;
             UserRole userRole = _currentUserService.Role;
 
-            var result = new List<AttachmentDto>();
-
-            if (userRole == UserRole.Student)
-            {
-                result = await _unitOfWork.AttachmentRepository
-                    .GetAttachmentList(accountId, null, null, accountId, request.StartDate, request.FinishDate);
-            }
-            else
+            switch (userRole)
             {
-                result = await _unitOfWork.AttachmentRepository
-                    .GetAttachmentList(accountId, request.CourseID, request.GroupID, accountId, request.StartDate, request.FinishDate);
+                case UserRole.Student:
+                    request.StudentAccountID = _unitOfWork.StudentRepository.GetAllAsync().Result.Find(x => x.AccountId == accountId).Id;
```

Comment:
1. Why are you loading ALL students? Wouldn't something like this work?
`var student = await _unitOfWork.StudentRepository.GetByIdAsync(_currentUserService.EntityId);`
2. Why are you doing that synchronously?

Reply:
updated